repose-aggregator/components/filters/rackspace-auth-user-filter/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala

```diff
@@ -187,12 +173,33 @@ class RackspaceAuthUserFilter @Inject()(configurationService: ConfigurationServi
     }
   }
 
-  def parseUserGroupFromInputStream(inputStream: InputStream, contentType: String, sessionIds: List[String]): Option[RackspaceAuthUserGroup] = {
-    sessionIds.map(HeaderValue).sortWith(_.quality > _.quality).toStream.flatMap({ header => Option(datastore.get(s"$ddKey:${header.value}").asInstanceOf[Option[RackspaceAuthUserGroup]]) }).headOption
-      .getOrElse(Option(configuration.getV20).flatMap(parseUsername(_, inputStream, contentType, username2_0JSON, username2_0XML)))
-      .orElse(Option(configuration.getV11).flatMap(parseUsername(_, inputStream, contentType, username1_1JSON, username1_1XML)))
+  val usernameForgotPassword2_0Xml: UsernameParsingFunction = { is =>
+    val xml = XML.load(is)
+    val username = (xml \\ "forgotPasswordCredentials" \ "@username").text
+    (None, Option(username).filterNot(_.isEmpty))
+  }
+
+  val usernameForgotPassword2_0Json: UsernameParsingFunction = { is =>
+    val json = Json.parse(is)
+    val possibleUsername = (json \ "RAX-AUTH:forgotPasswordCredentials" \ "username").validate[String]
+
+    possibleUsername match {
+      case wrappedUsername: JsSuccess[String] if wrappedUsername.get.trim.nonEmpty => (None, Some(wrappedUsername.get))
+      case failure: JsError =>
+        logger.debug("2.0 JSON parsing failure: {}", JsError.toFlatJson(failure))
+        (None, None)
+      case _ => (None, None)
+    }
   }
 
+  def parseUserGroupFromInputStream(uri: String, inputStream: InputStream, contentType: String, sessionIds: List[String]): Option[RackspaceAuthUserGroup] =
+    uri match {
+      case "/v2.0/users/RAX-AUTH/forgot-pwd" => Option(configuration.getV20).flatMap(parseUsername(_, inputStream, contentType, usernameForgotPassword2_0Json, usernameForgotPassword2_0Xml))
```

Comment:
Identity staging works if I include a trailing slash in the URL, so I'll add a functional test for it and update this line to handle it.